clang/lib/CodeGen/CodeGenModule.cpp

```diff
@@ -1570,6 +1574,10 @@ void CodeGenModule::GenOpenCLArgMetadata(llvm::Function *Fn,
               ? llvm::ConstantAsMetadata::get(CGF->Builder.getInt32(
                     SYCLBufferLocationAttr->getLocationID()))
               : llvm::ConstantAsMetadata::get(CGF->Builder.getInt32(-1)));
+
+      argESIMDAccPtrs.push_back(
+          llvm::ConstantAsMetadata::get(CGF->Builder.getInt1(
+              parm->getAttr<SYCLSimdAccessorPtrAttr>() != nullptr)));
```

Comment:
I think it will be a bit more optimal.
```suggestion
              parm->hasAttr<SYCLSimdAccessorPtrAttr>())));
```

Reply:
not sure. I want 0 or 1 is a clean input for getInt1. Non-null parm->getAttr<SYCLSimdAccessorPtrAttr>() is neither

Reply:
Sorry, I didn't get your statement. I thought that `parm->getAttr<SYCLSimdAccessorPtrAttr>() != nullptr` actually gives bool which is 0 or 1 as well as `hasAttr` function which also returns bool (https://clang.llvm.org/doxygen/classclang_1_1Decl.html#ae7a63b99398864d86f53afd8a03c359b).

Reply:
Sorry, you are right. hasAttr is OK - will change